src/main/java/de/tum/in/www1/artemis/config/migration/entries/MigrationEntry20220210_160300.java

```diff
@@ -0,0 +1,86 @@
+package de.tum.in.www1.artemis.config.migration.entries;
+
+import java.util.List;
+import java.util.Set;
+
+import org.springframework.stereotype.Component;
+
+import de.tum.in.www1.artemis.config.migration.MigrationEntry;
+import de.tum.in.www1.artemis.domain.*;
+import de.tum.in.www1.artemis.domain.enumeration.ProgrammingLanguage;
+import de.tum.in.www1.artemis.repository.ProgrammingExerciseRepository;
+import de.tum.in.www1.artemis.repository.ProgrammingExerciseTestCaseRepository;
+import de.tum.in.www1.artemis.service.programming.ProgrammingExerciseTestCaseService;
+
+/**
+ * This migration sets the enum value testCaseType for every existing test case depending on its name and exercise's programming language.
+ */
+@Component
+public class MigrationEntry20220210_160300 extends MigrationEntry {
+
+    private final ProgrammingExerciseRepository programmingExerciseRepository;
+
+    private final ProgrammingExerciseTestCaseRepository programmingExerciseTestCaseRepository;
+
+    private final ProgrammingExerciseTestCaseService programmingExerciseTestCaseService;
+
+    public MigrationEntry20220210_160300(ProgrammingExerciseRepository programmingExerciseRepository, ProgrammingExerciseTestCaseRepository programmingExerciseTestCaseRepository,
+            ProgrammingExerciseTestCaseService programmingExerciseTestCaseService) {
+        this.programmingExerciseRepository = programmingExerciseRepository;
+        this.programmingExerciseTestCaseRepository = programmingExerciseTestCaseRepository;
+        this.programmingExerciseTestCaseService = programmingExerciseTestCaseService;
+    }
+
+    /**
+     * Retrieves all programming exercises from the database and executes the processing method in batches of 80 exercises to prevent database timeouts
```

Comment:
Out of curiosity: Why exactly is this needed. Since the exercises are all processed sequentially either way and there is no specific database call between each batch, this should not really make a difference.
Am I missing something here?

Reply:
It is a relic from when the logic was different and I saved the entities after each batch of exercises to the DB. This was clearly not the best solution and after changing I forgot to remove this. Thanks for the mention, I changed it now.